Purchases/Misc/Obsoletions.swift

```diff
@@ -335,11 +335,16 @@ public extension Purchases {
      * - Parameter completion: A completion block that is called when the `SKPaymentDiscount` is returned.
      * If it was not successful, there will be an `Error`.
      */
-    @available(iOS, introduced: 12.2, obsoleted: 12.2, message: "Obtain StoreProductDiscount from StoreProduct")
-    @available(tvOS, introduced: 12.2, obsoleted: 12.2, message: "Obtain StoreProductDiscount from StoreProduct")
-    @available(watchOS, introduced: 6.2, obsoleted: 6.2, message: "Obtain StoreProductDiscount from StoreProduct")
-    @available(macOS, introduced: 10.14.4, obsoleted: 10.14.4, message: "Obtain StoreProductDiscount from StoreProduct")
-    @available(macCatalyst, introduced: 13.0, obsoleted: 13.0, message: "Obtain StoreProductDiscount from StoreProduct")
+    @available(iOS, introduced: 12.2, obsoleted: 12.2,
+               message: "Obtain StoreProductDiscount from StoreProduct or use checkPromotionalOfferEligibility")
+    @available(tvOS, introduced: 12.2, obsoleted: 12.2,
+               message: "Obtain StoreProductDiscount from StoreProduct or use checkPromotionalOfferEligibility")
+    @available(watchOS, introduced: 6.2, obsoleted: 6.2,
+               message: "Obtain StoreProductDiscount from StoreProduct or use checkPromotionalOfferEligibility")
+    @available(macOS, introduced: 10.14.4, obsoleted: 10.14.4,
+               message: "Obtain StoreProductDiscount from StoreProduct or use checkPromotionalOfferEligibility")
+    @available(macCatalyst, introduced: 13.0, obsoleted: 13.0,
+               message: "Obtain StoreProductDiscount from StoreProduct or use checkPromotionalOfferEligibility")
```

Comment:
this message seems a bit confusing to me, like these are alternative ways of getting the same result. 


How about:
> "Check eligibility for a discount using `checkPromotionalOfferEligibility:`, then pass in StoreProductDiscount from StoreProduct into purchase(package:discount:)"?

If that's too verbose, then I think the equivalent of `paymentDiscount(for:` is just `checkPromotionalOfferEligibility`, so I'd make this a rename instead of an obsoletion and have the message be 

> "Check eligibility for a discount using `checkPromotionalOfferEligibility:`"